packages/logger/src/format.ts

```diff
@@ -20,19 +20,41 @@ function isError(value: unknown): value is Error {
   )
 }
 
+/**
+ * Support errors with additional properties
+ * @shapeshiftoss/errors implement these additional properties
+ */
+function isErrorWithDetails(value: unknown): value is Error & {
+  code?: string
+  details?: Record<string, unknown>
+  cause: unknown
+} {
+  return isError(value) && 'cause' in value
```

Comment:
Should this be checking for `details`?

```suggestion
  return isError(value) && 'details' in value
```

Reply:
No because details is optional but `cause` always exists even if the value is undefined.